php/Terminus/Environment.php

```diff
@@ -358,26 +358,106 @@ public function mergeToDev($options = array()) {
     return $workflow;
   }
 
+  public function connectionInfo() {
+    $info = array();
+
+    // Can only SFTP into dev/multidev environments
+    if (!in_array($this->id, array('test', 'live'))) {
+      $sftp_username = sprintf("%s.%s", $this->id, $this->site->getId());
+      $sftp_password = "Use your account password";
+      $sftp_host = sprintf("appserver.%s.%s.drush.in", $this->id, $this->site->getId());
+      $sftp_port = 2222;
+
+      $sftp_url = sprintf('sftp://%s@%s:%s', $sftp_username, $sftp_host, $sftp_port);
+      $sftp_command = sprintf('sftp -o Port=%s %s@%s', $sftp_port, $sftp_username, $sftp_host);
+
+      $info = array_merge($info, array(
+        'sftp_username' => $sftp_username,
+        'sftp_host'     => $sftp_host,
+        'sftp_password' => $sftp_password,
+        'sftp_url'      => $sftp_url,
+        'sftp_command'  => $sftp_command
+      ));
+    }
 
-  /**
-   * Return the SFTP connection URL for this environment
-   */
-  public function sftp_url() {
-    $username = sprintf("%s.%s", $this->id, $this->site->getId());
-    $host = sprintf("appserver.%s.%s.drush.in", $this->id, $this->site->getId());
-    $port = 2222;
+    $git_username = sprintf("codeserver.%s.%s", $this->id, $this->site->getId());
+    $git_host = sprintf("codeserver.%s.%s.drush.in", $this->id, $this->site->getId());
+    $git_port = 2222;
 
-    return sprintf("sftp://%s@%s:%s", $username, $host, $port);
-  }
+    $git_url = sprintf("git://%s@%s:%s", $git_username, $git_host, $git_port);
+    $git_command = sprintf("git clone %s %s", $git_url, $this->site->getName());
 
-  /**
-   * Return the GIT connection URL for this environment
-   */
-  public function git_url() {
-    $username = sprintf("codeserver.dev.%s", $this->site->getId());
-    $host = sprintf("codeserver.dev.%s.drush.in", $this->site->getId());
-    $port = 2222;
-    return sprintf("git://%s@%s:%s", $username, $host, $port);
+    $info = array_merge($info, array(
+      'git_username' => $git_username,
+      'git_host'     => $git_host,
+      'git_port'     => $git_port,
+      'git_url'      => $git_url,
+      'git_command'  => $git_command
+    ));
+
+    $this->bindings->fetch();
+
+    $db_binding = @$this->bindings->getByType('dbserver')[0];
```

Comment:
`if(isset($this->bindings->getByType('dbserver')[0])) {
  $db_binding = $this->bindings->getByType('dbserver')[0];
}`